File: 0038-count-and-say/0038-count-and-say.cpp
class Solution {
public:
    string countAndSay(int n) {
        if (n==1) return "1";
        if (n==2) return "11";

        string s="11";
        for (int i=3; i<=n; i++) {
            string temp = "";
            s += "#";
            int cnt = 1;
            int m = s.size();
            for (int j=1; j<m; j++) {
               if ( s[j] != s[j-1]) {
                temp += to_string(cnt);
                temp += s[j-1];
                cnt = 1;
               }
               else {
                cnt ++;
               }
            }
            s = temp;
        }
        return s;
    }
};

File: 0038-count-and-say/0038-count-and-say.py
class Solution:
    def countAndSay(self, n: int) -> str:
        if n==1:
            return "1"
        if n==2:
            return "11"
        s = "11"
        for i in range(n-2):
            t = ""
            s += "#"
            c = 1
            # l=len(s)
            for j in range(1,len(s)):
                if s[j] != s[j-1]:
                    t += str(c)
                    t += s[j-1]
                    c = 1
                else :
                    c += 1
            s = t
        return s

        